src/main/java/org/pesmypetcare/webservice/controller/petmanager/PetMedicationRestController.java

```diff
@@ -0,0 +1,137 @@
+     * @param petName Name of the pet
+     * @param dateName Date of the Medication
+     * @return The MedicationEntity identified by the data
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/{owner}/{petName}/{dateName}")
+    public MedicationEntity getMedicationData(@PathVariable String owner, @PathVariable String petName,
+                                              @PathVariable String dateName)
+            throws DatabaseAccessException {
+        return medicationService.getMedicationData(owner, petName, dateName);
+    }
+
+    /**
+     * Gets the data from all the specified Medications from the database identified by its pet.
+     * @param owner Username of the owner of the pets
+     * @param petName Name of the pet
+     * @return The List containing all the Medications from the pet
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/{owner}/{petName}")
+    public List<Map<List<String>, Object>> getAllMedicationData(@PathVariable String owner,
+                                                                @PathVariable String petName)
+            throws DatabaseAccessException {
+        return medicationService.getAllMedicationData(owner, petName);
+    }
+
+    /**
+     * Gets the data from all the Medications eaten by the pet between the initial and final date not including them.
+     * @param owner Username of the owner of the pets
+     * @param petName Name of the pet
+     * @param initialDate Initial Date
+     * @param finalDate Final Date
+     * @return The List containing all the Medications eaten by the pet in the specified time
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/{owner}/{petName}/between/{initialDate}/{finalDate}")
+    public List<Map<List<String>, Object>>
+    getAllMedicationsBetween(@PathVariable String owner, @PathVariable String petName,
+                             @PathVariable String initialDate, @PathVariable String finalDate)
+            throws DatabaseAccessException {
+        return medicationService.getAllMedicationsBetween(owner, petName, initialDate, finalDate);
+    }
+
+    /**
+     * Gets the value for the specified field of the Medication on the database.
+     * @param owner Username of the owner of the pets
+     * @param petName Name of the pet
+     * @param dateName date + name of the Medication
+     * @param field Name of the field to retrieve the value from
+     * @return The value from the field on the database
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    @GetMapping("/{owner}/{petName}/{dateName}/{field}")
+    public Object getMedicationField(@PathVariable String owner, @PathVariable String petName,
+                                     @PathVariable String dateName,  @PathVariable String field)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a single space to separate non-whitespace characters.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)